rust/core-lib/src/plugins/manager.rs

```diff
@@ -110,43 +112,67 @@ impl <W: Write + Send + 'static>PluginManager<W> {
     pub fn start_plugin(&mut self, self_ref: &PluginManagerRef<W>,
                           view_id: &ViewIdentifier, plugin_name: &str, init_info: PluginBufferInfo) {
         //TODO: error handling: this should maybe have a completion callback with a Result
-        let key = (view_id.to_owned(), plugin_name.to_owned());
-        if self.running.contains_key(&key) {
-            return print_err!("plugin {} already running for buffer {}", plugin_name, view_id);
-        }
-
-        let plugin = self.catalog.iter().find(|desc| desc.name == plugin_name)
-            .expect(&format!("no plugin found with name {}", plugin_name));
-
-        let me = self_ref.clone();
-        plugin.launch(self_ref, view_id, move |result| {
-            match result {
-                Ok(plugin_ref) => {
-                    plugin_ref.initialize(&init_info);
-                    let mut inner = me.lock();
-                    inner.buffers.lock().editor_for_view(&key.0).unwrap().plugin_started(&key.0, &key.1);
-                    inner.running.insert(key, plugin_ref);
-                },
-                Err(_) => panic!("error handling is not implemented"),
+        let buffer_id = self.buffers.buffer_for_view(view_id);
+        if let Some(buffer_id) = buffer_id {
+            let key = (buffer_id.to_owned(), plugin_name.to_owned());
+            if self.running.contains_key(&key) {
+                return print_err!("plugin {} already running for buffer {}", plugin_name, key.0);
             }
-        });
+
+            let plugin = self.catalog.iter().find(|desc| desc.name == plugin_name)
+                .expect(&format!("no plugin found with name {}", plugin_name));
+
+            let me = self_ref.clone();
+            let view_id2 = view_id.to_owned();
+            plugin.launch(self_ref, view_id, move |result| {
+                match result {
+                    Ok(plugin_ref) => {
+                        plugin_ref.initialize(&init_info);
+                        let mut inner = me.lock();
+                        let mut running = false;
+
+                        if let Some(ed) = inner.buffers.lock()
+                            .editor_for_view(&view_id2) {
+                                ed.plugin_started(&view_id2, &key.1);
+                                running = true;
+                            }
```

Comment:
hmm, I was doing it this way because `inner.running.insert` is a mutable borrow, and we have an immutable borrow out when we take `editor_for_view`. It's possible I'm missing something. Just moving it into the `if` block doesn't compile.

My editor wants to align that closing brace with the second line of the opening expression. I don't really have an opinion here, will change, but it might get reverted next time I fix whitespace 😐.